akka-coordination/src/main/scala/akka/coordination/lease/scaladsl/LeaseProvider.scala

```diff
@@ -65,15 +64,17 @@ class LeaseProvider(system: ExtendedActorSystem) extends Extension {
     val fqcn = leaseSettings.leaseConfig.getString("lease-class")
     require(fqcn.nonEmpty, "lease-class must not be empty")
     val dynamicAccess = system.dynamicAccess
-    dynamicAccess
-      .createInstanceFor[Lease](
-        fqcn,
-        immutable.Seq((classOf[LeaseSettings], leaseSettings), (classOf[ExtendedActorSystem], system)))
-      .recoverWith {
-        case _: NoSuchMethodException ⇒
-          dynamicAccess.createInstanceFor[Lease](fqcn, immutable.Seq((classOf[LeaseSettings], leaseSettings)))
-
-      } match {
+    val instance: Try[Lease] = dynamicAccess.createInstanceFor[Lease](
+      fqcn,
+      immutable.Seq((classOf[LeaseSettings], leaseSettings), (classOf[ExtendedActorSystem], system))) match {
+      case s: Success[Lease] =>
+        s
+      case Failure(_: NoSuchMethodException) =>
+        dynamicAccess.createInstanceFor[Lease](fqcn, immutable.Seq((classOf[LeaseSettings], leaseSettings)))
+      case f: Failure[_] =>
+        f
+    }
```

Comment:
This is the actual fix, match instead of `recoverWith`? 👍

Reply:
yes, fix/workaround ;)